pkg/kubectl/cmd/create_secret.go

```diff
@@ -89,6 +89,7 @@ func NewCmdCreateSecretGeneric(f cmdutil.Factory, cmdOut io.Writer) *cobra.Comma
 	cmd.Flags().StringArray("from-literal", []string{}, "Specify a key and literal value to insert in secret (i.e. mykey=somevalue)")
 	cmd.Flags().String("from-env-file", "", "Specify the path to a file to read lines of key=val pairs to create a secret (i.e. a Docker .env file).")
 	cmd.Flags().String("type", "", i18n.T("The type of secret to create"))
+	cmd.Flags().Bool("append-hash", false, "Append a hash of the secret contents to its name.")
```

Comment:
Offline discussion: We can't think of a good reason to avoid supporting `--append-hash` for `tls` or `docker-registry`, but want to double check that it doesn't decrease security in those cases (e.g. possibility of a rainbow-table based attack against the `docker-registry` secrets).

@liggitt, thoughts on this?

Otherwise I'll extend it to those subcommands this week.